charts/datadog/values.yaml

```diff
@@ -624,6 +624,12 @@ datadog:
         name: otel-http
     # datadog.otelCollector.config -- OTel collector configuration
     config: null
+    # datadog.otelCollector.configMap -- Use an existing ConfigMap for DDOT Collector configuration
+    configMap:
+      # datadog.otelCollector.configMap.name -- Name of the existing ConfigMap that contains the DDOT Collector configuration
+      name: null
+      # datadog.otelCollector.configMap.key -- Key within the ConfigMap that contains the DDOT Collector configuration
+      key: otel-config.yaml
```

Comment:
This ensures that when key is unset, it will default to `otel-config.yaml ` right ? In the operator, we don't have support for key, and force users to set the key to `otel-config.yaml`, see https://github.com/DataDog/datadog-operator/blob/main/internal/controller/datadogagent/feature/otelcollector/feature.go#L193

Reply:
Correct, if the user will not specify ConfigMap key they want to use, it will default to `otel-config.yaml`.
Technically speaking, there's no reason to force user to use specific key as long as it maps internally to the correct path in the volume mount (`otel-config.yaml`). I think we should consider **updating the DD Operator implementation** to loosen this requirement and allow users to use whatever key they already have (or prefer to use).